editor/connections_dialog.cpp

```diff
@@ -788,23 +788,31 @@ bool ConnectionsDock::_is_item_signal(TreeItem &p_item) {
 }
 
 bool ConnectionsDock::_is_connection_inherited(Connection &p_connection) {
-	Node *scene_root = EditorNode::get_singleton()->get_edited_scene();
-	Ref<PackedScene> scn = ResourceLoader::load(scene_root->get_scene_file_path());
-	ERR_FAIL_NULL_V(scn, false);
-
-	Ref<SceneState> state = scn->get_state();
-	ERR_FAIL_NULL_V(state, false);
-
 	Node *source = Object::cast_to<Node>(p_connection.signal.get_object());
 	Node *target = Object::cast_to<Node>(p_connection.callable.get_object());
 
-	const NodePath source_path = scene_root->get_path_to(source);
-	const NodePath target_path = scene_root->get_path_to(target);
 	const StringName signal_name = p_connection.signal.get_name();
 	const StringName method_name = p_connection.callable.get_method();
 
-	// If it cannot be found in PackedScene, this connection was inherited.
-	return !state->has_connection(source_path, signal_name, target_path, method_name, true);
+	String file_path = source->get_scene_file_path().is_empty() ? source->get_owner()->get_scene_file_path() : source->get_scene_file_path();
+
+	const NodePath source_path = source->get_scene_file_path().is_empty() ? source->get_owner()->get_path_to(source) : NodePath(".");
+	const NodePath target_path = target->get_scene_file_path().is_empty() ? target->get_owner()->get_path_to(target) : NodePath(".");
+
+	Ref<PackedScene> scn = ResourceLoader::load(file_path);
+	ERR_FAIL_NULL_V(scn, false);
+
+	Ref<SceneState> state = scn->get_state();
+	ERR_FAIL_NULL_V(state, false);
+
+	bool exists_in_packed_scene_directly = state->has_connection(source_path, signal_name, target_path, method_name, true);
```

Comment:
I believe it'd make the code more readable without resorting to comments. This feels still pretty hack-ish to do

Reply:
It's more like, it'll likely be compiled out of the equation in your typical C++ compiler because it's just a never changing boolean. I can still remove it if it's way too beyond the general coding style.